docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/containerobject/dsl/ContainerNetworkObjectDslTestEnricher.java

```diff
@@ -0,0 +1,129 @@
+ */
+public class ContainerNetworkObjectDslTestEnricher implements TestEnricher {
+
+    private static final Logger logger = Logger.getLogger(ContainerNetworkObjectDslTestEnricher.class.getName());
+
+    @Inject
+    Instance<CubeController> cubeControllerInstance;
+
+    @Inject
+    Instance<CubeRegistry> cubeRegistryInstance;
+
+    @Inject
+    Instance<DockerClientExecutor> dockerClientExecutorInstance;
+
+    @Inject
+    Instance<NetworkRegistry> networkRegistryInstance;
+
+    @Inject
+    Instance<Injector> injectorInstance;
+
+    @Override
+    public void enrich(final Object testCase) {
+        // Needs to get definitions of network and contaienrs at same enricher because first we need to register networks and then containers
+        // And there is no way to order test enrichers.
+        startNetworks(testCase);
+        startContainers(testCase);
+
+    }
+
+    private void startContainers(Object testCase) {
+        final List<Field> containerFields = ReflectionUtil.getFieldsWithAnnotation(testCase.getClass(), DockerContainer.class);
+        Collections.sort(containerFields, Comparator.comparingInt(f -> f.getAnnotation(DockerContainer.class).order()));
+        Collections.reverse(containerFields);
+
+        for (Field field : containerFields) {
+            try {
+                final Object object = field.get(testCase);
+                if (Container.class.isAssignableFrom(object.getClass())) {
+
+                    Container containerObject = (Container) object;
+                    final Container enrichedContainer = injectorInstance.get().inject(containerObject);
+                    field.set(testCase, enrichedContainer);
+                    startContainer(enrichedContainer, testCase.getClass());
+
+                } else {
+                    throw new IllegalArgumentException(String.format("Object %s is not assignable to %s.", object.getClass(), Container.class.getName()));
+                }
+            } catch (IllegalAccessException e) {
+                throw new IllegalArgumentException(e);
+            }
+        }
+    }
+
+    private void startNetworks(Object testCase) {
+        List<Field> networkFields = ReflectionUtil.getFieldsWithAnnotation(testCase.getClass(), DockerNetwork.class);
+
+        for (Field field : networkFields) {
+            try {
+
+                final Object object = field.get(testCase);
+                if (Network.class.isAssignableFrom(object.getClass())) {
+                    Network networkObject = (Network) object;
+
+                    startNetwork(networkObject);
+
+                } else {
+                    throw new IllegalArgumentException(String.format("Object %s is not assignable to %s.", object.getClass(), Network.class.getName()));
+                }
+
+            } catch (IllegalAccessException e) {
+                throw new IllegalArgumentException(e);
```

Comment:
Why changing from one type to another here which are anyway not our own exceptions?